mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingLanguageDao.java

```diff
@@ -26,11 +30,38 @@ default HibLanguage create(String languageName, String languageTag, String uuid)
 	@Override
 	default LanguageResponse transformToRestSync(HibLanguage element, InternalActionContext ac, int level,
 			String... languageTags) {
+		GenericParameters generic = ac.getGenericParameters();
+		FieldsSet fields = generic.getFields();
+
 		LanguageResponse model = new LanguageResponse();
-		model.setUuid(element.getUuid());
-		model.setLanguageTag(element.getLanguageTag());
-		model.setName(element.getName());
-		model.setNativeName(element.getNativeName());
+		// TODO align fields.isEmpty() with other DAOs on a major update - this is breaking at the moment.
+		if (fields.isEmpty() || fields.has("uuid")) {
+			model.setUuid(element.getUuid());
+
+			// Performance shortcut to return now and ignore the other checks
+			if (fields.size() == 1) {
+				return model;
+			}
+		}
+		if (fields.isEmpty() || fields.has("name")) {
+			model.setName(element.getName());
+
+			// Performance shortcut to return now and ignore the other checks
+			if (fields.size() == 1) {
+				return model;
+			}
+		}
+		if (fields.isEmpty() || fields.has("tag")) {
```

Comment:
Wouldn't it be better to align the possible values for "fields" with the attribute names in the rest model? We should use "languageTag" here instead of "tag".